src/app/App.tsx

```diff
@@ -16,15 +19,29 @@ import { RoutesSwitch } from './RoutesSwitch';
 import { store } from './store';
 
 export const App = () => {
+  const { t } = useCustomTranslation();
+
+  const [isModalShow, setIsModalShow] = useState(false);
   const history = useHistory();
 
   const [reconnectRoom] = useMutation<ReconnectRoomQuery>(RECONNECT_ROOM);
+  const [leaveRoom] = useMutation(LEAVE_ROOM);
   const { data, loading, error } = useQuery<IsRoomExistQuery>(IS_ROOM_EXIST);
 
+  const leave = async () => {
+    try {
+      await leaveRoom();
+      setIsModalShow(false);
```

Comment:
Do we need to close modal in case of exception? If yes, it's better to place setIsModalShow(false) into `finally` block